genesyscloud/task_management_worktype/data_source_genesyscloud_task_management_worktype.go

```
@@ -40,3 +40,32 @@ func dataSourceTaskManagementWorktypeRead(ctx context.Context, d *schema.Resourc
 		return nil
 	})
 }
+
+func dataSourceTaskManagementWorktypeStatusRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getTaskManagementWorktypeProxy(sdkConfig)
+
+	workTypeName := d.Get("worktype_name").(string)
+	statusName := d.Get("worktype_status_name").(string)
+
+	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		//First retrieve the work type id by name
+		workType, retryable, err := proxy.getTaskManagementWorktypeByName(ctx, workTypeName)
+
+		if err != nil && !retryable {
+			return retry.NonRetryableError(fmt.Errorf("error searching task management worktype %s: %s", workTypeName, err))
+		}
+
+		if retryable {
+			return retry.RetryableError(fmt.Errorf("no task management worktype found with name %s", workTypeName))
+		}
+
+		for _, status := range *workType.Statuses {
+			if *status.Name == statusName {
+				d.SetId(*status.Id)
+				return nil
+			}
+		}
+		return nil
```

Shouldn't we return an error down here if there is no match on the status name?